
Unique values of features

age: continuous.

workclass: Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked.

fnlwgt: continuous.

education: Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool.

education-num: continuous.

marital-status: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.

occupation: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.

relationship: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.

race: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.

sex: Female, Male.

capital-gain: continuous.

capital-loss: continuous.

hours-per-week: continuous.

native-country: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands.

salary: >50K,<=50K

In [1]:
### Import Libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv("C:\\Users\\user\\Desktop\\2021 Dataset\\adult.csv")
df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [5]:
df.shape

(48842, 15)

In [6]:
df.columns

Index(['age', 'workclass', 'fnlwgt', 'education', 'educational-num',
       'marital-status', 'occupation', 'relationship', 'race', 'gender',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'income'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


In [8]:
###Statistics for numerical features

df.describe()

age        fnlwgt  educational-num  capital-gain  \
count  48842.000000  4.884200e+04     48842.000000  48842.000000   
mean      38.643585  1.896641e+05        10.078089   1079.067626   
std       13.710510  1.056040e+05         2.570973   7452.019058   
min       17.000000  1.228500e+04         1.000000      0.000000   
25%       28.000000  1.175505e+05         9.000000      0.000000   
50%       37.000000  1.781445e+05        10.000000      0.000000   
75%       48.000000  2.376420e+05        12.000000      0.000000   
max       90.000000  1.490400e+06        16.000000  99999.000000   

       capital-loss  hours-per-week  
count  48842.000000    48842.000000  
mean      87.502314       40.422382  
std      403.004552       12.391444  
min        0.000000        1.000000  
25%        0.000000       40.000000  
50%        0.000000       40.000000  
75%        0.000000       45.000000  
max     4356.000000       99.000000

In [9]:
###Statistics for non-numerical features

df.describe(include = ['object'])

workclass education      marital-status      occupation relationship  \
count      48842     48842               48842           48842        48842   
unique         9        16                   7              15            6   
top      Private   HS-grad  Married-civ-spouse  Prof-specialty      Husband   
freq       33906     15784               22379            6172        19716   

         race gender native-country income  
count   48842  48842          48842  48842  
unique      5      2             42      2  
top     White   Male  United-States  <=50K  
freq    41762  32650          43832  37155

In [11]:
###Check for missing values

df.isnull().sum()

age                0
workclass          0
fnlwgt             0
education          0
educational-num    0
marital-status     0
occupation         0
relationship       0
race               0
gender             0
capital-gain       0
capital-loss       0
hours-per-week     0
native-country     0
income             0
dtype: int64

In [38]:
###Drop rows with '?'

df = df.replace({'?': np.nan}).dropna()

In [40]:
df.shape

(45222, 15)

#### Question 1: How many men and women are represented in this dataset?

In [42]:
df['gender'].value_counts()

Male      30527
Female    14695
Name: gender, dtype: int64

In [43]:
df['gender'].value_counts(normalize=True)

Male      0.675048
Female    0.324952
Name: gender, dtype: float64

#### There are 30,527 (67.5%) Men and 14,695 (32.5%) Women 

#### Question 2:  What is the average age of women?

In [118]:
avg_age = df[df['gender'] == 'Female']['age'].mean()
print('The average age of women is {}'.format(avg_age))

The average age of women is 36.98475671997278


#### Question 3: What is the proportion of German citizens?

In [119]:
int((df['native-country'] == 'Germany').sum()) / df.shape[0]

0.004267834239971695

#### Question 4: What are mean value and standard deviation of the age of those who recieve more than 50K per year and those who receive less than 50K per year?

In [97]:
greater_salary = df.loc[df['income'] == '>50K', 'age']
lesser_salary = df.loc[df['income'] == '<=50K', 'age']

print(round(greater_salary.mean()), round(greater_salary.std()))
print(round(lesser_salary.mean()), round(lesser_salary.std()))

44 10
37 14


#### Question 5: Is it true that people who receive more than 50k have at least high school education? 

In [102]:
df.loc[df['income'] == '>50K', 'education'].unique()

array(['Assoc-acdm', 'Some-college', 'Prof-school', 'HS-grad', 'Masters',
       'Bachelors', 'Assoc-voc', 'Doctorate', '9th', '10th', '7th-8th',
       '11th', '5th-6th', '1st-4th', '12th', 'Preschool'], dtype=object)

#### From the result, we can see that people with salary >50k have better degrees than just high school education

#### Question 6: Display statistics of age for each race and each gender. 

In [103]:
for (race, gender), sub_df in df.groupby(['race', 'gender']):
    print("Race: {0}, gender: {1}".format(race, gender))
    print(sub_df['age'].describe())

Race: Amer-Indian-Eskimo, gender: Female
count    166.000000
mean      36.530120
std       12.733004
min       17.000000
25%       27.000000
50%       35.000000
75%       46.000000
max       80.000000
Name: age, dtype: float64
Race: Amer-Indian-Eskimo, gender: Male
count    269.000000
mean      36.271375
std       10.636671
min       17.000000
25%       29.000000
50%       35.000000
75%       43.000000
max       71.000000
Name: age, dtype: float64
Race: Asian-Pac-Islander, gender: Female
count    436.000000
mean      35.979358
std       12.341018
min       17.000000
25%       26.000000
50%       35.000000
75%       44.000000
max       81.000000
Name: age, dtype: float64
Race: Asian-Pac-Islander, gender: Male
count    867.000000
mean      39.182238
std       12.669963
min       18.000000
25%       30.000000
50%       38.000000
75%       47.000000
max       90.000000
Name: age, dtype: float64
Race: Black, gender: Female
count    2084.000000
mean       38.100288
std        12.480038
min  

#### Question 7: What is the maximum number of hours a person works per week and How many people work such a number of hours?

In [113]:
max_work_hr = df['hours-per-week'].max()
workaholics = df[df['hours-per-week'] == max_work_hr].shape[0]
print('The maximum number of hours a person works per week is {}hrs/week'.format(max_work_hr))
print('{0} people work for {1}hrs/week'.format(workaholics, max_work_hr))

The maximum number of hours a person works per week is 99hrs/week
123 people work for 99hrs/week


#### Question 8: Count the average time of work for those earning a little and a lot of income for each country.

In [117]:
pd.crosstab(df['native-country'], df['income'], values=df['hours-per-week'], aggfunc=np.mean).T

native-country   Cambodia     Canada      China   Columbia       Cuba  \
income                                                                  
<=50K           41.294118  38.475728  37.753247  39.538462  39.919192   
>50K            43.888889  46.933333  42.027778  56.250000  42.852941   

native-country  Dominican-Republic    Ecuador  El-Salvador    England  France  \
income                                                                          
<=50K                    41.717391  39.081081    36.014706  40.722222   41.35   
>50K                     42.800000  47.833333    43.454545  46.297872   46.50   

native-country  ...  Portugal  Puerto-Rico   Scotland      South     Taiwan  \
income          ...                                                           
<=50K           ...     41.28    38.741935  40.833333  43.518072  37.066667   
>50K            ...     46.75    39.900000  50.000000  51.000000  45.760000   

native-country   Thailand  Trinadad&Tobago  United-States    Vietnam  \
income                                                                 
<=50K           44.333333            38.75      39.369245  38.618421   
>50K            53.000000            40.00      45.718989  39.428571   

native-country  Yugoslavia  
income                      
<=50K            38.533333  
>50K             43.375000  

[2 rows x 41 columns]